docs/configuration/initialization.md

```diff
@@ -0,0 +1,129 @@
+
+For example to execute something on the "callback" action when the request is a POST method, you can check for `req.query.nextauth.includes("callback") && req.mehod === "POST"`
+:::
+
+:::note
+`NextAuth` will implicitly close the response (by calling `res.end`, `res.send` or similar), so you should not run code **after** `NextAuth` in the function body. Using `return NextAuth` makes sure you don't forget that.
+:::
+
+Any variable you create this way will be available in the `NextAuth` options as well, since they are in the same scope.
+
+```ts title="/pages/api/auth/[...nextauth].ts"
+import type { NextApiRequest, NextApiResponse } from "next"
+import NextAuth from "next-auth"
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+
+  if(req.query.nextauth.includes("callback") && req.method === "POST") {
+    console.log(
+      "Handling callback request from my Identity Provider",
+      req.body
+    )
+  }
+
+  // Get a custom cookie value from the request
+  const someCookie = req.cookies["some-custom-cookie"]
+
+  return await NextAuth(req, res, {
+    ...
+    callbacks: {
+      session({ session, token }) {
+        // Return a cookie value as part of the session
+        // This is read when `req.query.nextauth.includes("session") && req.method === "GET"`
+        session.someCookie = someCookie
+        return session
+      }
+    }
+  })
+}
+```
+
+A practical example could be to not show a certain provider on the default sign-in page, but still be able to sign in with it. (The idea is taken from [this discussion](https://github.com/nextauthjs/next-auth/discussions/3133)):
+
+```js title="/pages/api/auth/[...nextauth].js"
+import NextAuth from "next-auth"
+import CredentialsProvider from "next-auth/providers/credentials"
+import GoogleProvider from "next-auth/providers/google"
+
+export default async function auth(req, res) {
+  const providers = [
+    CredentialsProvider(...),
+    GoogleProvider(...),
+  ]
+
+  const isDefaultSigninPage = req.method === "GET" && req.query.nextauth.includes("signin")
+
+  // Will hide the `GoogleProvider` when you visit `/api/auth/signin`
+  if (isDefaultSigninPage) providers.pop()
+
+  return await NextAuth(req, res, {
+    providers,
+    ...
+  })
+}
+```
+
+To completely understand what actions correspond to and what methods are supported, check out the [REST API documentation](/getting-started/rest-api), or look at the [source code](https://github.com/nextauthjs/next-auth/blob/beta/src/core/index.ts).
+
+This way of initializing `NextAuth` is very powerful, but should be used sparingly.
+
+:::warning
+Changing parts of the request that is essential to `NextAuth` to do it's job - like messing with the [default cookies](/configuration/options#cookies) - can have unforeseen consequences, and have the potential to introduce security holes if done incorrectly. Only change those this if you understand these consequences.
```

Comment:
Good idea with the warning at the end here! "This method of initializing NextAuth.js is very powerful, so if you fuck it up its on you ;)" :joy: